src/mappings/entities/mapAsset.ts

```diff
@@ -0,0 +1,436 @@
+  );
+
+const excludeTransferManager = (
+  manager: AssetAdvancedCompliance,
+  managers: AssetAdvancedCompliance[]
+) =>
+  managers.filter(
+    e =>
+      e.CountTransferManager !== manager.CountTransferManager ||
+      e.PercentageTransferManager !== manager.PercentageTransferManager
+  );
+// #endregion
+
+// #region ModuleIdEnum.Asset
+const handleCreateAsset = async (params: Record<string, any>, extrinsic: any) => {
+  const { name, ticker, assetType: type, fundingRound, divisible, disableIu, identifiers } = params;
+  const ownerDid = extrinsic.events[1].event.data[0].toString();
+  await Asset.create({
+    id: ticker,
+    ticker,
+    name,
+    type,
+    fundingRound,
+    isDivisible: divisible,
+    isFrozen: false,
+    isUniquenessRequired: !disableIu,
+    documents: [],
+    identifiers: formatAssetIdentifiers(identifiers),
+    ownerDid,
+    fullAgents: [ownerDid],
+    totalSupply: '0',
+    totalTransfers: '0',
+    compliance: { isPaused: false, sender: [], receiver: [], advanced: [] },
+  }).save();
+};
+
+const handleRenameAsset = async (params: Record<string, any>) => {
+  const { ticker, name: newName } = params;
+  const asset = await getAsset(ticker);
+  asset.name = newName;
+  await asset.save();
+};
+
+const handleSetFundingRound = async (params: Record<string, any>) => {
+  const { ticker, name: newFundingRound } = params;
+  const asset = await getAsset(ticker);
+  asset.fundingRound = newFundingRound;
+  await asset.save();
+};
+
+type AssetNewDocument = {
+  name: string;
+  uri: string;
+  content_hash?: any;
+  doc_type?: string;
+  filing_date?: Date;
+};
+
+const handleAddDocuments = async (params: Record<string, any>, extrinsic: any) => {
+  const { ticker, docs } = params;
+  const asset = await getAsset(ticker);
+  asset.documents = docs.map((doc: AssetNewDocument, i: number) => ({
+    id: Number(extrinsic.events[i].event.data[2].toString()),
+    name: doc.name,
+    link: doc.uri,
+  }));
+  await asset.save();
+};
+
+const handleRemoveDocuments = async (params: Record<string, any>) => {
+  const { ticker, ids } = params;
+  const asset = await getAsset(ticker);
+  asset.documents = asset.documents.filter(doc => !ids.includes(doc.id));
+  await asset.save();
+};
+
+const handleUpdateIdentifiers = async (params: Record<string, any>) => {
+  const { ticker, identifiers: newIdentifiers } = params;
+  const asset = await getAsset(ticker);
+  asset.identifiers = formatAssetIdentifiers(newIdentifiers);
+  await asset.save();
+};
+
+const handleMakeDivisible = async (params: Record<string, any>) => {
+  const { ticker } = params;
+  const asset = await getAsset(ticker);
+  asset.isDivisible = true;
+  await asset.save();
+};
+
+const handleIssue = async (params: Record<string, any>) => {
+  const { ticker, amount } = params;
+  const asset = await getAsset(ticker);
+  const formattedAmount = chainAmountToBigNumber(amount);
+  const assetOwner = await AssetHolder.get(`${asset.ownerDid}:${ticker}`);
```

Comment:
This isn't guaranteed to be found.

I think this needs to be wrapped with a `findOrCreate` type method.